src/System.Private.ServiceModel/tests/Common/Infrastructure/src/ConditionalTestDetectors.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace Infrastructure.Common
+{
+    // This class contains helper "detector" methods to detect specific
+    // conditions about the environment in which the tests are run.
+    // They are used for [ConditionalFact] and [ConditionalTheory]
+    // tests.  They are called at most once, and the return value cached.
+    internal static class ConditionalTestDetectors
+    {
+        public static bool IsRootCertificateInstalled()
+        {
+            try
+            {
+                ServiceUtilHelper.EnsureRootCertificateInstalled();
+                return true;
```

Comment:
I know we called `EnsureRootCertificateInstalled()` here, but I would like us to still keep the detection logic as you had in your previous PR. Let the detection tell if the installation is successful or not instead of making assumptions. In the future, in case we want to factor out installation logic at other places, it will be very easy too.